sklearn/pipeline.py

```diff
@@ -491,6 +491,25 @@ def decision_function(self, X):
             Xt = transform.transform(Xt)
         return self.steps[-1][-1].decision_function(Xt)
 
+    @if_delegate_has_method(delegate='_final_estimator')
+    def score_samples(self, X):
+        """Apply transforms, and score_samples of the final estimator.
+
+        Parameters
+        ----------
+        X : iterable
+            Data to predict on. Must fulfill input requirements of first step
+            of the pipeline.
+
+        Returns
+        -------
+        y_score : array-like, shape (n_samples, n_classes)
```

Comment:
score_samples does work with density estimators and many more than are not multiclass problems.
To me it should be:

`y_score : array-like, shape (n_samples,)`

Reply:
Sure. I have updated.
The same remark seems to apply to the functions `decision_function`, `predict_log_proba`and `predict_proba`. Should I also update their documentation ?